packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/useFileUploader.tsx

```diff
@@ -1,19 +1,73 @@
+import { checkMaxSize, setAcceptedFiles } from '@aws-amplify/ui';
 import React, { useState } from 'react';
-import { Files } from '../../types';
+import { Files, FileStatuses } from '../../types';
 import { UseFileUploader } from './types';
 
-export default function useFileUploader(): UseFileUploader {
+export default function useFileUploader(
+  maxSize: number,
+  acceptedFileTypes: string[],
+  multiple: boolean
+): UseFileUploader {
+  const [fileStatuses, setFileStatuses] = useState<FileStatuses>([]);
   const [showPreviewer, setShowPreviewer] = useState(false);
   const [files, setFiles] = useState<Files>([]);
 
   const [inDropZone, setInDropZone] = useState(false);
 
-  const addTargetFiles = (targetFiles: FileList) => {
+  const setFileSizeErrors = (files: Files, fileStatuses: FileStatuses) => {
+    const statuses = [...fileStatuses];
+    [...files].forEach((file, index) => {
+      const errorFile = checkMaxSize(maxSize, file);
+      const status = fileStatuses[index];
+
+      statuses[index] = {
+        ...status,
+        error: status?.error || !!errorFile,
+        fileErrors: status?.fileErrors ?? errorFile,
+        loading: false,
+      };
+    });
+    setFileStatuses(statuses);
+  };
+
+  const checkAndSetFiles = (targets: Files, statuses: FileStatuses) => {
+    setFileSizeErrors(targets, statuses);
+    setFiles(targets);
+  };
+
+  const addTargetFiles = (targetFiles: FileList): number => {
```

Comment:
This could probably use some refactoring, but there are two places where the `addTargetFiles` function is ran. One is in the `onDrop` handler, the other is in the `onFileChange`. One occurs when a button is clicked the other for drop.

I added in the return number to see if there was any files to be added after the `acceptedFileType` check. If the number was zero, then the `showPreviewer` would not show.

Reply:
I could move the logic to show the previewer in `addTargetFiles` but then it's mutating even more state, and maybe it shouldn't? Or maybe I should rename the function? I'm up for suggestions.